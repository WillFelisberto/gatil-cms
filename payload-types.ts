/* tslint:disable */
/* eslint-disable */
/**
 * This file was automatically generated by Payload.
 * DO NOT MODIFY IT BY HAND. Instead, modify your source Payload config,
 * and re-run `payload generate:types` to regenerate this file.
 */

/**
 * Supported timezones in IANA format.
 *
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "supportedTimezones".
 */
export type SupportedTimezones =
  | 'Pacific/Midway'
  | 'Pacific/Niue'
  | 'Pacific/Honolulu'
  | 'Pacific/Rarotonga'
  | 'America/Anchorage'
  | 'Pacific/Gambier'
  | 'America/Los_Angeles'
  | 'America/Tijuana'
  | 'America/Denver'
  | 'America/Phoenix'
  | 'America/Chicago'
  | 'America/Guatemala'
  | 'America/New_York'
  | 'America/Bogota'
  | 'America/Caracas'
  | 'America/Santiago'
  | 'America/Buenos_Aires'
  | 'America/Sao_Paulo'
  | 'Atlantic/South_Georgia'
  | 'Atlantic/Azores'
  | 'Atlantic/Cape_Verde'
  | 'Europe/London'
  | 'Europe/Berlin'
  | 'Africa/Lagos'
  | 'Europe/Athens'
  | 'Africa/Cairo'
  | 'Europe/Moscow'
  | 'Asia/Riyadh'
  | 'Asia/Dubai'
  | 'Asia/Baku'
  | 'Asia/Karachi'
  | 'Asia/Tashkent'
  | 'Asia/Calcutta'
  | 'Asia/Dhaka'
  | 'Asia/Almaty'
  | 'Asia/Jakarta'
  | 'Asia/Bangkok'
  | 'Asia/Shanghai'
  | 'Asia/Singapore'
  | 'Asia/Tokyo'
  | 'Asia/Seoul'
  | 'Australia/Brisbane'
  | 'Australia/Sydney'
  | 'Pacific/Guam'
  | 'Pacific/Noumea'
  | 'Pacific/Auckland'
  | 'Pacific/Fiji';

export interface Config {
  auth: {
    users: UserAuthOperations;
  };
  blocks: {};
  collections: {
    cats: Cat;
    adoptions: Adoption;
    users: User;
    guardians: Guardian;
    media: Media;
    sponsorships: Sponsorship;
    cronLogs: CronLog;
    'activity-log': ActivityLog;
    'payload-locked-documents': PayloadLockedDocument;
    'payload-preferences': PayloadPreference;
    'payload-migrations': PayloadMigration;
  };
  collectionsJoins: {};
  collectionsSelect: {
    cats: CatsSelect<false> | CatsSelect<true>;
    adoptions: AdoptionsSelect<false> | AdoptionsSelect<true>;
    users: UsersSelect<false> | UsersSelect<true>;
    guardians: GuardiansSelect<false> | GuardiansSelect<true>;
    media: MediaSelect<false> | MediaSelect<true>;
    sponsorships: SponsorshipsSelect<false> | SponsorshipsSelect<true>;
    cronLogs: CronLogsSelect<false> | CronLogsSelect<true>;
    'activity-log': ActivityLogSelect<false> | ActivityLogSelect<true>;
    'payload-locked-documents':
      | PayloadLockedDocumentsSelect<false>
      | PayloadLockedDocumentsSelect<true>;
    'payload-preferences': PayloadPreferencesSelect<false> | PayloadPreferencesSelect<true>;
    'payload-migrations': PayloadMigrationsSelect<false> | PayloadMigrationsSelect<true>;
  };
  db: {
    defaultIDType: string;
  };
  globals: {
    'site-config': SiteConfig;
  };
  globalsSelect: {
    'site-config': SiteConfigSelect<false> | SiteConfigSelect<true>;
  };
  locale: 'pt';
  user: User & {
    collection: 'users';
  };
  jobs: {
    tasks: unknown;
    workflows: unknown;
  };
}
export interface UserAuthOperations {
  forgotPassword: {
    email: string;
    password: string;
  };
  login: {
    email: string;
    password: string;
  };
  registerFirstUser: {
    email: string;
    password: string;
  };
  unlock: {
    email: string;
    password: string;
  };
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "cats".
 */
export interface Cat {
  id: string;
  nome: string;
  idade?: string | null;
  descricao?: string | null;
  sexo: 'M' | 'F';
  foto?: (string | null) | Media;
  galeria?:
    | {
        imagem?: (string | null) | Media;
        legenda?: string | null;
        id?: string | null;
      }[]
    | null;
  vacinas?:
    | {
        nome: string;
        data?: string | null;
        observacoes?: string | null;
        id?: string | null;
      }[]
    | null;
  vermifugacoes?:
    | {
        nome?: string | null;
        data?: string | null;
        observacoes?: string | null;
        id?: string | null;
      }[]
    | null;
  castrado?: boolean | null;
  doencas?: string | null;
  dataEntrada?: string | null;
  observacoesSaude?: string | null;
  adotado?: boolean | null;
  show?: boolean | null;
  disponivelParaApadrinhamento?: boolean | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "media".
 */
export interface Media {
  id: string;
  alt?: string | null;
  updatedAt: string;
  createdAt: string;
  url?: string | null;
  thumbnailURL?: string | null;
  filename?: string | null;
  mimeType?: string | null;
  filesize?: number | null;
  width?: number | null;
  height?: number | null;
  focalX?: number | null;
  focalY?: number | null;
  sizes?: {
    thumbnail?: {
      url?: string | null;
      width?: number | null;
      height?: number | null;
      mimeType?: string | null;
      filesize?: number | null;
      filename?: string | null;
    };
    card?: {
      url?: string | null;
      width?: number | null;
      height?: number | null;
      mimeType?: string | null;
      filesize?: number | null;
      filename?: string | null;
    };
    tablet?: {
      url?: string | null;
      width?: number | null;
      height?: number | null;
      mimeType?: string | null;
      filesize?: number | null;
      filename?: string | null;
    };
  };
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "adoptions".
 */
export interface Adoption {
  id: string;
  gato: string | Cat;
  tutor: string | Guardian;
  dataAdocao: string;
  observacoes?: string | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "guardians".
 */
export interface Guardian {
  id: string;
  nome: string;
  email: string;
  telefone: string;
  document?: string | null;
  endereco?: string | null;
  whatsapp?: string | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "users".
 */
export interface User {
  id: string;
  name: string;
  telefone?: string | null;
  showPhone?: boolean | null;
  photo?: (string | null) | Media;
  role: 'admin' | 'voluntario';
  show?: boolean | null;
  updatedAt: string;
  createdAt: string;
  email: string;
  resetPasswordToken?: string | null;
  resetPasswordExpiration?: string | null;
  salt?: string | null;
  hash?: string | null;
  loginAttempts?: number | null;
  lockUntil?: string | null;
  password?: string | null;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "sponsorships".
 */
export interface Sponsorship {
  id: string;
  gato?: (string | null) | Cat;
  apadrinhaProjeto?: boolean | null;
  padrinho: string | Guardian;
  formaPagamento?: ('Pix' | 'Cartão' | 'Dinheiro') | null;
  valorMensal: 'R$ 29,90 (Básico)' | 'R$ 59,90 (Premium)' | 'R$ 99,90 (Master)';
  certificadoDigital?: boolean | null;
  visitas?: boolean | null;
  aceitaTermos: boolean;
  dataAtualizacao?: string | null;
  proximaAtualizacao?: string | null;
  ativo?: boolean | null;
  whatsapp?: string | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "cronLogs".
 */
export interface CronLog {
  id: string;
  executadoEm?: string | null;
  totalVencidos?: number | null;
  emailEnviado?: boolean | null;
  mensagem?: string | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "activity-log".
 */
export interface ActivityLog {
  id: string;
  user?: {
    relationTo: 'users';
    value: string | User;
  } | null;
  operation?: ('create' | 'read' | 'update' | 'delete') | null;
  timestamp?: string | null;
  ipAddress?: string | null;
  deviceInfo?: string | null;
  locale?: string | null;
  resource?: string | null;
  documentId?: string | null;
  data?:
    | {
        [k: string]: unknown;
      }
    | unknown[]
    | string
    | number
    | boolean
    | null;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "payload-locked-documents".
 */
export interface PayloadLockedDocument {
  id: string;
  document?:
    | ({
        relationTo: 'cats';
        value: string | Cat;
      } | null)
    | ({
        relationTo: 'adoptions';
        value: string | Adoption;
      } | null)
    | ({
        relationTo: 'users';
        value: string | User;
      } | null)
    | ({
        relationTo: 'guardians';
        value: string | Guardian;
      } | null)
    | ({
        relationTo: 'media';
        value: string | Media;
      } | null)
    | ({
        relationTo: 'sponsorships';
        value: string | Sponsorship;
      } | null)
    | ({
        relationTo: 'cronLogs';
        value: string | CronLog;
      } | null)
    | ({
        relationTo: 'activity-log';
        value: string | ActivityLog;
      } | null);
  globalSlug?: string | null;
  user: {
    relationTo: 'users';
    value: string | User;
  };
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "payload-preferences".
 */
export interface PayloadPreference {
  id: string;
  user: {
    relationTo: 'users';
    value: string | User;
  };
  key?: string | null;
  value?:
    | {
        [k: string]: unknown;
      }
    | unknown[]
    | string
    | number
    | boolean
    | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "payload-migrations".
 */
export interface PayloadMigration {
  id: string;
  name?: string | null;
  batch?: number | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "cats_select".
 */
export interface CatsSelect<T extends boolean = true> {
  nome?: T;
  idade?: T;
  descricao?: T;
  sexo?: T;
  foto?: T;
  galeria?:
    | T
    | {
        imagem?: T;
        legenda?: T;
        id?: T;
      };
  vacinas?:
    | T
    | {
        nome?: T;
        data?: T;
        observacoes?: T;
        id?: T;
      };
  vermifugacoes?:
    | T
    | {
        nome?: T;
        data?: T;
        observacoes?: T;
        id?: T;
      };
  castrado?: T;
  doencas?: T;
  dataEntrada?: T;
  observacoesSaude?: T;
  adotado?: T;
  show?: T;
  disponivelParaApadrinhamento?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "adoptions_select".
 */
export interface AdoptionsSelect<T extends boolean = true> {
  gato?: T;
  tutor?: T;
  dataAdocao?: T;
  observacoes?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "users_select".
 */
export interface UsersSelect<T extends boolean = true> {
  name?: T;
  telefone?: T;
  showPhone?: T;
  photo?: T;
  role?: T;
  show?: T;
  updatedAt?: T;
  createdAt?: T;
  email?: T;
  resetPasswordToken?: T;
  resetPasswordExpiration?: T;
  salt?: T;
  hash?: T;
  loginAttempts?: T;
  lockUntil?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "guardians_select".
 */
export interface GuardiansSelect<T extends boolean = true> {
  nome?: T;
  email?: T;
  telefone?: T;
  document?: T;
  endereco?: T;
  whatsapp?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "media_select".
 */
export interface MediaSelect<T extends boolean = true> {
  alt?: T;
  updatedAt?: T;
  createdAt?: T;
  url?: T;
  thumbnailURL?: T;
  filename?: T;
  mimeType?: T;
  filesize?: T;
  width?: T;
  height?: T;
  focalX?: T;
  focalY?: T;
  sizes?:
    | T
    | {
        thumbnail?:
          | T
          | {
              url?: T;
              width?: T;
              height?: T;
              mimeType?: T;
              filesize?: T;
              filename?: T;
            };
        card?:
          | T
          | {
              url?: T;
              width?: T;
              height?: T;
              mimeType?: T;
              filesize?: T;
              filename?: T;
            };
        tablet?:
          | T
          | {
              url?: T;
              width?: T;
              height?: T;
              mimeType?: T;
              filesize?: T;
              filename?: T;
            };
      };
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "sponsorships_select".
 */
export interface SponsorshipsSelect<T extends boolean = true> {
  gato?: T;
  apadrinhaProjeto?: T;
  padrinho?: T;
  formaPagamento?: T;
  valorMensal?: T;
  certificadoDigital?: T;
  visitas?: T;
  aceitaTermos?: T;
  dataAtualizacao?: T;
  proximaAtualizacao?: T;
  ativo?: T;
  whatsapp?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "cronLogs_select".
 */
export interface CronLogsSelect<T extends boolean = true> {
  executadoEm?: T;
  totalVencidos?: T;
  emailEnviado?: T;
  mensagem?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "activity-log_select".
 */
export interface ActivityLogSelect<T extends boolean = true> {
  user?: T;
  operation?: T;
  timestamp?: T;
  ipAddress?: T;
  deviceInfo?: T;
  locale?: T;
  resource?: T;
  documentId?: T;
  data?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "payload-locked-documents_select".
 */
export interface PayloadLockedDocumentsSelect<T extends boolean = true> {
  document?: T;
  globalSlug?: T;
  user?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "payload-preferences_select".
 */
export interface PayloadPreferencesSelect<T extends boolean = true> {
  user?: T;
  key?: T;
  value?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "payload-migrations_select".
 */
export interface PayloadMigrationsSelect<T extends boolean = true> {
  name?: T;
  batch?: T;
  updatedAt?: T;
  createdAt?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "site-config".
 */
export interface SiteConfig {
  id: string;
  /**
   * Número do WhatsApp utilizado no site para adoção/contato no geral
   */
  whatsapp?: string | null;
  links?:
    | {
        type: 'instagram' | 'facebook' | 'tiktok';
        url: string;
        id?: string | null;
      }[]
    | null;
  updatedAt?: string | null;
  createdAt?: string | null;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "site-config_select".
 */
export interface SiteConfigSelect<T extends boolean = true> {
  whatsapp?: T;
  links?:
    | T
    | {
        type?: T;
        url?: T;
        id?: T;
      };
  updatedAt?: T;
  createdAt?: T;
  globalType?: T;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "auth".
 */
export interface Auth {
  [k: string]: unknown;
}

declare module 'payload' {
  export interface GeneratedTypes extends Config {}
}
