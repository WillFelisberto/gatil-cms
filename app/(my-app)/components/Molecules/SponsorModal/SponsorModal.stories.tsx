/* eslint-disable no-alert */
import type { Meta, StoryObj } from '@storybook/react';

import { Cat } from '@/payload-types';

import { SponsorModal } from './SponsorModal';

const mockCat: Cat = {
  id: 1,
  nome: 'Gato Teste',
  sexo: 'M',
  idade: '1 ano',
  descricao: 'maluca doidinha',
  createdAt: '2025-04-02T21:24:52.909Z',
  updatedAt: '2025-04-03T20:30:41.034Z',
  foto: {
    createdAt: '2025-04-02T20:48:51.116Z',
    updatedAt: '2025-04-02T20:48:51.116Z',
    alt: 'Will',
    filename: 'gato3.png',
    mimeType: 'image/png',
    filesize: 1008532,
    width: 800,
    height: 800,
    focalX: 50,
    focalY: 50,
    sizes: {
      thumbnail: {
        width: 400,
        height: 300,
        mimeType: 'image/png',
        filesize: 324915,
        filename: 'gato3-400x300.png',
        url: '/api/media/file/gato3-400x300.png'
      },
      card: {
        width: 768,
        height: 1024,
        mimeType: 'image/png',
        filesize: 1794522,
        filename: 'gato3-768x1024.png',
        url: '/api/media/file/gato3-768x1024.png'
      },
      tablet: {
        width: null,
        height: null,
        mimeType: null,
        filesize: null,
        filename: null,
        url: null
      }
    },
    id: 23121,
    url: '/api/media/file/gato3.png',
    thumbnailURL: '/api/media/file/gato3-400x300.png'
  },
  galeria: [],
  vacinas: [],
  vermifugacoes: [],
  castrado: true,
  dataEntrada: '2025-04-02T12:00:00.000Z',
  adotado: false,
  show: true,
  disponivelParaApadrinhamento: true
};

const meta: Meta<typeof SponsorModal> = {
  title: 'Molecules/SponsorModal',
  component: SponsorModal,
  tags: ['autodocs']
};

export default meta;
type Story = StoryObj<typeof SponsorModal>;

export const Default: Story = {
  args: {
    cat: mockCat,
    onClose: () => alert('Fechar modal')
  }
};
