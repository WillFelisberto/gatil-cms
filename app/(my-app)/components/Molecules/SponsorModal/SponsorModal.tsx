'use client';

import { X } from 'lucide-react';
import Image from 'next/image';
import { useEffect } from 'react';

import { Cat, Media } from '@/payload-types';

import { SponsorForm } from '../../Atoms/SponsorForm';

type SponsorModalProps = {
  cat: Cat;
  onClose(): void;
};

export const SponsorModal = ({ cat, onClose }: SponsorModalProps) => {
  // Fecha no ESC
  useEffect(() => {
    const handleKeyDown = (e: KeyboardEvent) => {
      if (e.key === 'Escape') onClose();
    };
    document.addEventListener('keydown', handleKeyDown);
    return () => document.removeEventListener('keydown', handleKeyDown);
  }, [onClose]);

  return (
    <div
      className="fixed inset-0 z-50 bg-black/60 flex items-center justify-center px-4"
      role="dialog"
      aria-modal="true"
      aria-labelledby="sponsor-modal-title"
      data-testid="sponsor-modal"
    >
      <div className="relative bg-white rounded-xl shadow-xl w-full max-w-2xl max-h-[95vh] overflow-y-auto p-6">
        {/* Bot√£o de fechar */}
        <button
          onClick={onClose}
          aria-label="Fechar modal"
          className="absolute top-4 right-4 text-gray-600 hover:text-red-600 transition"
          data-testid="close-button"
        >
          <X className="w-5 h-5" />
        </button>

        <h2
          id="sponsor-modal-title"
          className="text-2xl font-bold text-[#013274] mb-4"
          data-testid="modal-title"
        >
          Apadrinhar {cat.nome}
        </h2>

        {/* Imagem do gato */}
        {cat.foto && typeof cat.foto === 'object' && (
          <div className="relative w-full h-56 mb-6 rounded-lg overflow-hidden shadow">
            <Image
              src={(cat.foto as Media).url || ''}
              alt={(cat.foto as Media).alt || `Foto de ${cat.nome}`}
              fill
              className="object-cover"
              sizes="(max-width: 768px) 100vw, 640px"
              data-testid={`cat-image-${cat.nome}`}
            />
          </div>
        )}

        {cat.descricao && (
          <p className="text-sm text-gray-700 mb-6" data-testid="cat-description">
            {cat.descricao}
          </p>
        )}

        <SponsorForm cat={cat} onSuccess={onClose} />
      </div>
    </div>
  );
};
