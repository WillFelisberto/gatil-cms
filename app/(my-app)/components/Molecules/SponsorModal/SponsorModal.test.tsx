import { Cat, Media } from '@/payload-types';
import { fireEvent, render, screen } from '@/tests/test-utils';

import { SponsorModal } from './SponsorModal';

const mockCatWithImage: Cat = {
  id: '1',
  nome: 'Gato Teste',
  sexo: 'M',
  createdAt: '2025-04-02T21:24:52.909Z',
  updatedAt: '2025-04-03T20:30:41.034Z',
  idade: '1 ano',
  descricao: 'maluca doidinha',
  foto: {
    id: '123',
    url: '/gato.jpg',
    alt: 'Foto linda',
    filename: 'gato.jpg',
    mimeType: 'image/png',
    filesize: 1000,
    width: 800,
    height: 600,
    createdAt: '',
    updatedAt: '',
    sizes: {},
    thumbnailURL: ''
  } as Media,
  galeria: [],
  vacinas: [],
  vermifugacoes: [],
  castrado: true,
  dataEntrada: '2025-04-02T12:00:00.000Z',
  adotado: false,
  show: true,
  disponivelParaApadrinhamento: true
};

const mockCatWithoutImage = { ...mockCatWithImage, foto: undefined } as Cat;
const mockCatWithStringImage = {
  ...mockCatWithImage,
  foto: {
    createdAt: '2025-04-02T20:48:51.116Z',
    updatedAt: '2025-04-02T20:48:51.116Z',
    alt: 'Will',
    filename: '/gato.jpg',
    mimeType: 'image/png',
    filesize: 1008532,
    width: 800,
    height: 800,
    focalX: 50,
    focalY: 50,
    sizes: {
      thumbnail: {
        width: 400,
        height: 300,
        mimeType: 'image/png',
        filesize: 324915,
        filename: 'gato3-400x300.png',
        url: 'gato3-400x300.png'
      },

      card: {
        width: 768,
        height: 1024,
        mimeType: 'image/png',
        filesize: 1794522,
        filename: 'gato3-768x1024.png',
        url: 'gato3-768x1024.png'
      },

      tablet: {
        width: null,
        height: null,
        mimeType: null,
        filesize: null,
        filename: null,
        url: null
      }
    },
    id: '67eda2b3b852bb6b7ae49216',
    url: '/gato.jpg',
    thumbnailURL: 'gato3-400x300.png'
  }
};
const mockCatWithoutDescription = { ...mockCatWithImage, descricao: undefined } as Cat;

describe('<SponsorModal />', () => {
  const onCloseMock = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with image and description', () => {
    const { container } = render(<SponsorModal cat={mockCatWithImage} onClose={onCloseMock} />);

    expect(screen.getByTestId('sponsor-modal')).toBeInTheDocument();
    expect(screen.getByTestId('modal-title')).toHaveTextContent(/Apadrinhar Gato Teste/i);
    expect(screen.getByTestId('cat-description')).toHaveTextContent(/maluca doidinha/i);
    expect(screen.getByTestId('cat-image-Gato Teste')).toBeInTheDocument();
    expect(screen.getByTestId('close-button')).toBeInTheDocument();

    expect(screen.getByRole('dialog')).toHaveAttribute('aria-modal', 'true');
    expect(screen.getByRole('dialog')).toHaveAttribute('aria-labelledby', 'sponsor-modal-title');

    expect(container).toMatchSnapshot();
  });

  it('calls onClose when clicking close button', () => {
    render(<SponsorModal cat={mockCatWithImage} onClose={onCloseMock} />);
    fireEvent.click(screen.getByTestId('close-button'));
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  it('calls onClose when pressing Escape', () => {
    render(<SponsorModal cat={mockCatWithImage} onClose={onCloseMock} />);
    fireEvent.keyDown(document, { key: 'Escape' });
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  it('does not render image if image is missing', () => {
    render(<SponsorModal cat={mockCatWithoutImage} onClose={onCloseMock} />);
    expect(screen.queryByRole('img')).not.toBeInTheDocument();
  });

  it('renders correctly with image as string URL', () => {
    render(<SponsorModal cat={mockCatWithStringImage} onClose={onCloseMock} />);
    const img = screen.getByTestId(`cat-image-${mockCatWithImage.nome}`) as HTMLImageElement;

    expect(img.src).toContain('gato.jpg');
  });

  it('does not render description if missing', () => {
    render(<SponsorModal cat={mockCatWithoutDescription} onClose={onCloseMock} />);
    expect(screen.queryByTestId('cat-description')).not.toBeInTheDocument();
  });

  it('renders correct image attributes', () => {
    render(<SponsorModal cat={mockCatWithImage} onClose={onCloseMock} />);
    const img = screen.getByTestId(`cat-image-${mockCatWithImage.nome}`) as HTMLImageElement;

    expect(img.alt).toBe('Foto linda');
    expect(img.src).toContain('gato.jpg');
  });
});
