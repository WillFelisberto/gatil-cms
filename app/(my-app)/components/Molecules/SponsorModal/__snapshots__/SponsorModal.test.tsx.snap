// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SponsorModal /> renders correctly with image and description 1`] = `
<div>
  <div
    aria-labelledby="sponsor-modal-title"
    aria-modal="true"
    class="fixed inset-0 z-50 bg-black/60 flex items-center justify-center px-4"
    data-testid="sponsor-modal"
    role="dialog"
  >
    <div
      class="relative bg-white rounded-xl shadow-xl w-full max-w-2xl max-h-[95vh] overflow-y-auto p-6"
    >
      <button
        aria-label="Fechar modal"
        class="absolute top-4 right-4 text-gray-600 hover:text-red-600 transition"
        data-testid="close-button"
      >
        <svg
          class="lucide lucide-x w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </button>
      <h2
        class="text-2xl font-bold text-[#013274] mb-4"
        data-testid="modal-title"
        id="sponsor-modal-title"
      >
        Apadrinhar 
        Gato Teste
      </h2>
      <div
        class="relative w-full h-56 mb-6 rounded-lg overflow-hidden shadow"
      >
        <img
          alt="Foto linda"
          class="object-cover"
          data-nimg="fill"
          data-testid="cat-image-Gato Teste"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, 640px"
          src="/_next/image?url=%2Fgato.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fgato.jpg&w=640&q=75 640w, /_next/image?url=%2Fgato.jpg&w=750&q=75 750w, /_next/image?url=%2Fgato.jpg&w=828&q=75 828w, /_next/image?url=%2Fgato.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fgato.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fgato.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fgato.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fgato.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <p
        class="text-sm text-gray-700 mb-6"
        data-testid="cat-description"
      >
        maluca doidinha
      </p>
      <form
        aria-label="Formulário de apadrinhamento"
        class="space-y-6 bg-white w-full mx-auto"
        data-testid="sponsor-form"
      >
        <div>
          <label
            class="block mb-1 font-semibold text-gray-800"
            for="padrinhoNome"
          >
            Seu nome
          </label>
          <input
            aria-invalid="false"
            class="w-full p-2 border rounded-md focus:ring-2 focus:outline-none border-gray-300"
            data-testid="input-nome"
            id="padrinhoNome"
            name="padrinhoNome"
            placeholder="Ex: João Vítor"
            type="text"
          />
        </div>
        <div>
          <label
            class="block mb-1 font-semibold text-gray-800"
            for="whatsapp"
          >
            WhatsApp
          </label>
          <input
            aria-invalid="false"
            class="w-full p-2 border rounded-md focus:ring-2 focus:outline-none border-gray-300"
            data-testid="input-whatsapp"
            id="whatsapp"
            name="whatsapp"
            placeholder="(00) 00000-0000"
            value=""
          />
        </div>
        <div>
          <label
            class="block mb-1 font-semibold text-gray-800"
            for="padrinhoEmail"
          >
            Seu email
          </label>
          <input
            aria-invalid="false"
            class="w-full p-2 border rounded-md focus:ring-2 focus:outline-none border-gray-300"
            data-testid="input-email"
            id="padrinhoEmail"
            name="padrinhoEmail"
            placeholder="seuemail@exemplo.com"
            type="email"
          />
        </div>
        <div
          class="flex flex-col sm:flex-row sm:items-start sm:gap-4"
        >
          <label
            class="font-semibold text-gray-800 min-w-[180px]"
            for="valorMensal"
          >
            Plano de contribuição
          </label>
          <div
            class="flex flex-col flex-1"
          >
            <select
              aria-invalid="false"
              class="p-2 border rounded-md focus:ring-2 focus:outline-none border-gray-300"
              data-testid="select-plano"
              id="valorMensal"
              name="valorMensal"
            >
              <option
                value=""
              >
                Selecione um plano
              </option>
              <option
                value="R$ 29,90 (Básico)"
              >
                R$ 29,90 (Básico)
              </option>
              <option
                value="R$ 59,90 (Premium)"
              >
                R$ 59,90 (Premium)
              </option>
              <option
                value="R$ 99,90 (Master)"
              >
                R$ 99,90 (Master)
              </option>
            </select>
          </div>
        </div>
        <div
          class="flex flex-col sm:flex-row sm:items-start sm:gap-4"
        >
          <label
            class="font-semibold text-gray-800 min-w-[180px]"
            for="formaPagamento"
          >
            Forma de pagamento
          </label>
          <div
            class="flex flex-col flex-1"
          >
            <select
              aria-invalid="false"
              class="p-2 border rounded-md focus:ring-2 focus:outline-none border-gray-300"
              data-testid="select-pagamento"
              id="formaPagamento"
              name="formaPagamento"
            >
              <option
                value=""
              >
                Selecione
              </option>
              <option
                value="Pix"
              >
                Pix
              </option>
              <option
                value="Dinheiro"
              >
                Dinheiro
              </option>
            </select>
          </div>
        </div>
        <div
          class="space-y-3"
        >
          <label
            class="flex items-center gap-2 text-gray-700"
          >
            <input
              data-testid="checkbox-certificado"
              name="certificadoDigital"
              type="checkbox"
            />
            Deseja receber certificado digital?
          </label>
          <label
            class="flex items-center gap-2 text-gray-700"
          >
            <input
              data-testid="checkbox-visitas"
              name="visitas"
              type="checkbox"
            />
            Gostaria de participar de eventos ou visitas?
          </label>
          <label
            class="flex items-center gap-2 text-gray-700"
          >
            <input
              aria-invalid="false"
              data-testid="checkbox-termos"
              name="aceitaTermos"
              type="checkbox"
            />
            Li e aceito os termos de apadrinhamento
          </label>
        </div>
        <button
          class="w-full bg-[#013274] text-white font-semibold py-2 rounded-md hover:bg-[#012050] transition"
          data-testid="submit-button"
          type="submit"
        >
          Apadrinhar agora
        </button>
      </form>
    </div>
  </div>
</div>
`;
